core/src/main/scala/codesearch/core/index/repository/Extensions.scala

```diff
@@ -3,25 +3,29 @@ package codesearch.core.index.repository
 import simulacrum.typeclass
 
 @typeclass trait Extensions[A] {
-  def extensions: Set[String]
+  def commonExtensions: Set[String] = Set("json", "md", "txt", "xml", "yml", "yaml", "properties")
+  def sourceExtensions: Set[String]
```

Comment:
docstrings :trollface: 

seriously, the logic here and especially the justification for it (why do we need two sets of extensions?) is not trivial so there should be a comment and maybe even a link to the issue